content/en/tracing/guide/instrument_custom_method.md

```diff
@@ -0,0 +1,353 @@
+                self.ledger[transaction.id] = transaction
+
+        # [...]
+```
+
+[1]: https://docs.datadoghq.com/tracing/guide/add_span_md_and_graph_it
+{{% /tab %}}
+{{% tab "Ruby" %}}
+  In Ruby, Datadog APM allows you to instrument your code to generate custom spans by instrumenting specific code blocks.
+
+  This example creates a new span for every transaction posted to the ledger and adds a [custom tag][1] with the specific transaction ID to the span.
+
+```ruby
+require 'ddtrace'
+
+class BackupLedger
+
+  def write(transactions)
+    # Use global `Datadog.tracer.trace` to trace blocks of inline code
+    Datadog.tracer.trace('BackupLedger.write') do |method_span|
+      transactions.each do |transaction|
+        Datadog.tracer.trace('BackupLedger.persist') do |span|
+          # Add custom metadata to the "persist_transaction" span
+          span.set_tag('transaction.id', transaction.id)
+          ledger[transaction.id] = transaction
+        end
+      end
+    end
+
+    # [...]
+  end
+end
+```
+[1]: https://docs.datadoghq.com/tracing/guide/add_span_md_and_graph_it
+{{% /tab %}}
+{{% tab "Go" %}}
+  In Go, Datadog APM allows you to instrument your code to generate custom spans by instrumenting specific code blocks.
+
+  This example creates a new span for every transaction posted to the ledger and adds a [custom tag][1] with the specific transaction ID to the span.
+
+```go
+package ledger
+
+import "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+// [...]
+
+func (bl *BackupLedger) write(ctx context.Context, transactions []*Transaction) (err error) {
+  // Trace the `write` function and capture the error if present
+  span, ctx := tracer.StartSpanFromContext(ctx, "BackupLedger.write")
+  defer func() {
+    span.Finish(tracer.WithError(err))
+  }()
+
+  for _, t := range transactions {
+    if err := bl.persistTransaction(ctx, t); err != nil {
+      return err
+    }
+  }
+  return nil
+}
+
+// persistTransaction is an inner function you may want to Trace. You can use the
+// same approach as before because the `ctx` you pass down includes out-of-the-box span
+// references to create a parent/child relationship
+func (bl *BackupLedger) persistTransaction(ctx context.Context, transaction *Transaction) error {
+  id := transaction.ID
+  span, _ := tracer.StartSpanFromContext(ctx, "BackupLedger.persist", tracer.Tag("transaction_id", id))
+  defer span.Finish()
+
+  if t, ok := bl.transactions[id]; ok {
+    return errors.New("duplicate entry")
+  }
+  bl.transactions[id] = transaction
+  return nil
+}
+```
+
+[1]: https://docs.datadoghq.com/tracing/guide/add_span_md_and_graph_it
+{{% /tab %}}
+{{% tab "Node.js" %}}
+  In Node.js, Datadog APM allows you to instrument your code to generate custom spans by instrumenting specific code blocks.
+
+This example creates a new span for every transaction posted to the ledger and adds a [custom tag][1] with the specific transaction ID to the span.
```

Comment:
We probably want to adjust this sentence like for Ruby as well, call out what we are adding a parent span with child spans (which have a tag set).

Reply:
Done!